### PR TITLE
Hw 21

### PR DESCRIPTION
**Pull request by KrohIgor**

### CODE REVIEW

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products";
+        List<Product> productList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                long productId = resultSet.getLong("product_id");
+                String productName = resultSet.getString("productname");
```
Comment (boroda4436) [inline src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: code duplication on line 43

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                Double productPrice = resultSet.getDouble("productprice");
+                Product product = new Product(productName, productPrice);
+                product.setProductId(productId);
+                productList.add(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException();
```
Comment (boroda4436) [inline src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]:
```suggestion
            throw new DataProcessingException(e);
```
Do not lose helpful info, re throw the exception

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        String query = "UPDATE `internet-shop`.`products` SET `productname` = ?, "
+                + "`productprice` = ? WHERE (`product_id` = ?)";
+        try (Connection connection = ConnectionUtil.getConnection();) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, product.getName());
+            preparedStatement.setDouble(2, product.getPrice());
+            preparedStatement.setLong(3, product.getProductId());
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            throw new DataProcessingException();
```
Comment (boroda4436) [inline src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]:
```suggestion
            throw new DataProcessingException(e);
```

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM `internet-shop`.`products` WHERE (`product_id` = ?);";
+        try (Connection connection = ConnectionUtil.getConnection();) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            preparedStatement.execute();
+            return true;
+        } catch (SQLException e) {
+            throw new DataProcessingException();
```
Comment (boroda4436) [inline src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]:
```suggestion
            throw new DataProcessingException(e);
```

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            if (resultSet.next()) {
+                long productId = resultSet.getLong("product_id");
+                String productName = resultSet.getString("productname");
+                Double productPrice = resultSet.getDouble("productprice");
+                Product product = new Product(productName, productPrice);
+                product.setProductId(productId);
+                return Optional.of(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException();
```
Comment (boroda4436) [inline src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]:
```suggestion
            throw new DataProcessingException(e);
```

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 1 above)
Comment (KrohIgor) [reply src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: take out in a separate method

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 3 above)
Comment (KrohIgor) [reply src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 2 above)
Comment (KrohIgor) [reply src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 4 above)
Comment (KrohIgor) [reply src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 5 above)
Comment (KrohIgor) [reply src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        String query = "SELECT * FROM products WHERE product_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getProductFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(e);
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: also it's better to add custom message throw new DataProcessingException("Couldn't get Product with id - " + id,  e);

File: src/main/resources/init_db.sql [hunk 7]
```diff
@@ -0,0 +1,6 @@
+CREATE SCHEMA `internet-shop` DEFAULT CHARACTER SET utf8 ;
+CREATE TABLE `internet-shop`.`products` (
+  `product_id` BIGINT(11) NOT NULL AUTO_INCREMENT,
+  `productname` VARCHAR(225) NOT NULL,
+  `productprice` DOUBLE NOT NULL,
+  PRIMARY KEY (`product_id`));
```
Comment (dmytro-hryhoruk) [inline src/main/resources/init_db.sql]: empty line

**Review by dmytro-hryhoruk — APPROVED** [review]

File: src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 6 above)
Comment (KrohIgor) [reply src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/resources/init_db.sql (hunk 7 above)
Comment (KrohIgor) [reply src/main/resources/init_db.sql]: done

**Pull request merged**